src/libtrx/config/common.c

```diff
@@ -26,32 +51,14 @@ bool Config_Read(void)
     const bool result = ConfigFile_Read(&args);
     if (result) {
         Config_Sanitize();
-        Config_ApplyChanges();
+        Config_ApplyChanges(CCT_INITIAL_READ);
     }
     return result;
 }
 
 bool Config_Write(void)
 {
-    Config_Sanitize();
-    const CONFIG_IO_ARGS args = {
-        .default_path = Config_GetPath(CFT_DEFAULT),
-        .enforced_path = Config_GetPath(CFT_ENFORCED),
-        .action = &Config_DumpToJSON,
-    };
-    const bool updated = ConfigFile_Write(&args);
-    if (updated) {
-        Config_ApplyChanges();
-        if (m_EventManager != NULL) {
-            const EVENT event = {
-                .name = "write",
-                .sender = NULL,
-                .data = NULL,
-            };
-            EventManager_Fire(m_EventManager, &event);
-        }
-    }
-    return updated;
+    return M_Write(CCT_WRITE);
```

Comment:
Maybe it's needed for a future PR so forgive the question, but do we need to extract this code out to `M_Write` if only `Config_Write` calls it?

Reply:
Good call – I initially had another call that used this, but it was no longer needed. Let me bring it back.

Reply:
Actually I think it's sufficient to just reorder the calls, and we can forgo the trigger changes altogether simplifying the final solution.